### PR TITLE
FIX: Restructure Item Delegates

### PR DESCRIPTION
**Pull request by zdomke**
Restructure ItemDelegates to make an abstract class to manage the editor_list. Added functionality to reset the editor_list on model reset.

### CODE REVIEW

File: archive_viewer/widgets/item_delegates.py [hunk 1]
```diff
@@ -5,11 +5,42 @@
...
-class SliderDelegate(QStyledItemDelegate):
+class EditorDelegate(QStyledItemDelegate):
+    def __init__(self, parent: QTableView) -> None:
+        super().__init__(parent)
+        self.editor_list = []
+        model = self.parent().model()
+        model.modelAboutToBeReset.connect(self.reset_editors)
+
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: QModelIndex) -> None:
+        """Create a new persistent editor on the Table View at the given index."""
```
Comment (YektaY) [inline archive_viewer/widgets/item_delegates.py:19]: can you add the return and parameters to the doc for the methods added by this PR? Thanks

**Review by YektaY — COMMENTED** [review]
This looks good! Would just request that the doc stings have the parameters and returns added to them.

File: archive_viewer/widgets/item_delegates.py (hunk 1 above)
Comment (zdomke) [reply archive_viewer/widgets/item_delegates.py]: I updated the docstrings and return parameters

**Comment by zdomke** [comment]
Additional changes:
1. Added abstractmethods to EditorDelegate
2. Removed the unused delegates SliderDelegate and CheckboxDelegate
  a. This also included removing the now unused widget CenterCheckbox (deprecated in #16)

**Review by YektaY — APPROVED** [review]

**Pull request merged**